src/register_users/shared_secret_register/v1.rs

```diff
@@ -0,0 +1,80 @@
+//! [POST /_synapse/admin/v1/register](https://matrix-org.github.io/synapse/latest/admin_api/register_api.html#shared-secret-registration)
+
+use ruma::{api::ruma_api, OwnedDeviceId, OwnedServerName, OwnedUserId};
+use serde::{Deserialize, Serialize};
+
+ruma_api! {
+    metadata: {
+        description: "Shared-secret registration",
+        method: POST,
+        name: "shared_secret_register_v1",
+        unstable_path: "/_synapse/admin/v1/register",
+        rate_limited: false,
+        authentication: None,
+    }
+
+    #[derive(Serialize)]
```

Comment:
This isn't how requests are meant to be constructed. Ruma provides traits for (de)serializing requests and responses, and the `ruma_api!` macro implements those already, properly accounting for path fields, query parameters and setting the path.

Reply:
How would the correct usage look like? And what kind of refactorings are you expecting for the PR?

Reply:
Correct usage would be to go through [`ruma::client::Client::send_request`](https://docs.rs/ruma/latest/ruma/client/struct.Client.html#method.send_request), `matrix_sdk::Client::send` or (lower level) [`OutgoingRequest::try_into_http_request`](https://docs.rs/ruma/latest/ruma/api/trait.OutgoingRequest.html#tymethod.try_into_http_request) for serialization plus [`IncomingResponse::try_from_http_response`](https://docs.rs/ruma/latest/ruma/api/trait.IncomingResponse.html#tymethod.try_from_http_response).

For this PR, I just want you to remove any `Serialize` and `Deserialize` derives on `request` / `response` blocks.

Reply:
Thanks for the explanation! I was looking for the `OutgoingRequest` implementation in the locally generated docs, but probably just overlooked them.

Also adjusted the PR by removing de-/serialize.